### ISSUE TITLE
Filename fix for viertausendhertz.de 

### ISSUE THREAD

**Issue by charakterziffer** [dvehrs/podget#14]
I gave podget the RSS feed http://viertausendhertz.de/feed/podcast/systemfehler. The enclosures there look like http://viertausendhertz.de/podcast-download/4511/sf04.mp3?v=1470947681&source=feed

With all file name fixes enabled but the Nº 5 (default setting) this will be fixed as “sf04_v_1470947681_038_source_feed.mp3”. What setting do I need to get the simple file name “sf04.mp3”?

More comprehensive: Do you plan to restructure the file name fixes?, e.g. to
F1) Should parts of the query string be reused in the file name? – default: yes (1)
F1.1) Limit this to the following parameters (e.g. guid, media_url, episodeName, "" [values without a parameter]; empty list means preserve the whole query string)
F2) Should the number part in the URL be reused in the file name? – default: no (2)
F2.2) Maybe: Limit this to the following domains (e.g. theonion.com, soundcloud.com)
F3) Should remaining bad chars be fixed? – default: yes (1)
F3.1) List of bad chars (e.g. "`~!#$^&=+{}*[]:;\"'<>?|\")

So the present file name fixes will be included in these three steps: 2,3,8,11 in F1 – 4,6,7,9,10 after F1.1 – 5,9 in F2 – 1 in F3 (which is at the end, at least in the process of fixing). The advantage would be that one wouldn’t need a new fix for every feed. But I’m not sure if this could be implemented practically.

**Comment by dvehrs** [comment]
I'm not sure there is a current setting that will simplify the filename.  I can see why that would be desirable but I've generally stuck with longer filenames so that they remain unique.

As for restructuring filenames, that is a good idea.  What we currently have is something that evolved as a series of single fixes for problems that users reported.  It definitely needs to evolve from what it is into something more unified and easier to maintain.

I do like how you've divided the process up into a logical order.  I'm not sure how to implement it either but you've started me thinking.  Sometime in the near future, we will need to fork a branch off the development branch to experiment with ideas.

Also it would be nice to have these new fixes enabled by an option on the line in the serverlist file as I've done with other recent changes.   That way we can get away from needing to run multiple sessions with different format fixing options.

Thanks for the ideas,

Dave

**Comment by charakterziffer** [comment]
Thank you for your response! Yes, a fix option on the line in the serverlist file may be the best solution. This way one can minimise contradictionary rules. All the best with the future development of this great podcatcher!

**Comment by bingalls** [comment]
@charakterziffer I'll bet you can remove **&source=feed** for now.

**Comment by charakterziffer** [comment]
@bingalls Could you give me a hint how to do it? I installed the newest podget version (0.8.5), removed my configuration file, adapted the newly created configuration file and used the FILENAME_FORMATFIX9=1. (“This is a fix for podcast feeds formatted like those for Audioboo. Removes everything after the ? in the filename. Attempted to make this fix generic enough to work with a variety of feeds of mp3, mp4, ogg and ogv files.”)

Running podget with the verbosity level 4 I get
ORIGINAL FILENAME:	sf24.mp3?v=1520584261&#038;source=feed
MODIFIED FILENAME:	sf24.mp3_v_1520584261_038_source_feed

That’s not exactly what I wanted. Am I doing something wrong? Do I have to set the filename fix in a way in the serverlist file? Do I need a certain combination of filename fixes? I only activated Fix9.

Sadly my code knowledge is not that profound to understand the fix in the podget file (line 1088ff):

```
if [[ -n ${FILENAME_FORMATFIX9+set} ]] && (( FILENAME_FORMATFIX9 > 0 )); then
   if (( $(${EXPR} "${MODIFIED_FILENAME}" : "[-_0-9A-z]\+[.][gmopv34]\+[?][%&;=0-9a-zA-Z]\+") > 0 )) ; then
      MODIFIED_FILENAME=$(echo "${MODIFIED_FILENAME}" | ${SED} -ru 's/([-_A-Za-z0-9]+[.][ogmpv3-4]+)[?][%&;=A-Za-z0-9]+/\1/g')
      if (( DEBUG == 1 )) ; then
         echo "${DEBUG_LEADER} FILENAME FORMAT(9) FIXED: ${MODIFIED_FILENAME}"
      fi
   fi
fi
```
The regular expression `'s/([-_A-Za-z0-9]+[.][ogmpv3-4]+)[?][%&;=A-Za-z0-9]+/\1/g'`should just keep the part before the ?, if I understand that correctly. Why does it seem not to be executed?

**Comment by bingalls** [comment]
@charakterziffer  Sorry, I could not get it working. While the last query parameter does not look to be required, removing it made no difference in my limited test.

**Comment by dvehrs** [comment]
OK, I've taken a look at it and while FILENAME_FORMATFIX9 **should** have fixed it, it wasn't being called due to an error in the EXPR line.  So I fixed that.  Then I discovered a minor issue also in the sed statement for the MODIFIED_FILENAME.  So I fixed that.

Now when I run podget with FILENAME_FORMATFIX9 enabled, I get filenames like "sf24.mp3" for this feed.  I've uploaded my proposed changes into the dev branch and if you'd like to test it feel free to download it.  All you will need is the podget file, so you don't have to download everything unless you want to.

Dave
